File: docs/current_docs/ci/quickstart/snippets/publish/java/src/main/java/io/dagger/modules/hellodagger/HelloDagger.java
package io.dagger.modules.hellodagger;

import io.dagger.client.Container;
import io.dagger.client.DaggerQueryException;
import io.dagger.client.Directory;
import io.dagger.client.CacheVolume;
import io.dagger.module.annotation.DefaultPath;
import io.dagger.module.annotation.Function;
import io.dagger.module.annotation.Object;
import java.util.List;
import java.util.concurrent.ExecutionException;

/** HelloDagger main object */
@Object
public class HelloDagger {
  /** Publish the application container after building and testing it on-the-fly */
  @Function
  public String publish(@DefaultPath("/") Directory source)
      throws InterruptedException, ExecutionException, DaggerQueryException {
    // call Dagger Function to run unit tests
    this.test(source);
    // call Dagger Function to build the application image
    // publish the image to ttl.sh
    return this.build(source).
        publish("ttl.sh/hello-dagger-%d".formatted((int) (Math.random() * 10000000)));
  }
}

File: docs/versioned_docs/version-0.17/ci/quickstart/snippets/publish/php/src/HelloDagger.php
<?php

declare(strict_types=1);

namespace DaggerModule;

use Dagger\Attribute\DaggerFunction;
use Dagger\Attribute\DaggerObject;
use Dagger\Attribute\DefaultPath;
use Dagger\Attribute\Doc;
use Dagger\Container;
use Dagger\Directory;

use function Dagger\dag;

#[DaggerObject]
class HelloDagger
{
    #[DaggerFunction]
    #[Doc('Publish the application container after building and testing it on-the-fly')]
    public function publish(
      #[DefaultPath('/')]
      Directory $source,
    ): string {
        // call Dagger Function to run unit tests
        $this->test($source);

        // call Dagger Function to build the application image
        // publish the image to ttl.sh
        return $this
            ->build($source)
            ->publish('ttl.sh/hello-dagger-' . rand(0, 10000000));
    }
}
